banzai/photometry.py

```diff
@@ -23,95 +26,125 @@ def radius_of_contour(contour, source):
     x_center = (source['xmax'] - source['xmin'] + 1) / 2.0 - 0.5
     y_center = (source['ymax'] - source['ymin'] + 1) / 2.0 - 0.5
 
-    return np.percentile(np.sqrt((x - x_center)**2.0 + (y - y_center)** 2.0), 90)
+    return np.percentile(np.sqrt((x - x_center)**2.0 + (y - y_center) ** 2.0), 90)
+
+
+def flag_sources(sources, source_labels, segmentation_map, mask, flag, mask_value):
+    affected_sources = np.unique(segmentation_map.data[mask == mask_value])
+    sources['flag'][np.in1d(source_labels, affected_sources)] |= flag
+
+
+def flag_deblended(sources, catalog, segmentation_map, deblended_seg_map, flag_value=2):
+    # By default deblending appends labels instead of reassigning them so we can just use the
+    # extras in the deblended map
+    deblended_sources = np.unique(deblended_seg_map.data[deblended_seg_map > np.max(segmentation_map)])
+    # Get the sources that were originally blended
+    original_blends = np.unique(segmentation_map.data[deblended_seg_map > np.max(segmentation_map)])
+    deblended_sources = np.hstack([deblended_sources, original_blends])
+    sources['flag'][np.in1d(catalog.labels, deblended_sources)] |= flag_value
+
+
+def flag_edge_sources(image, sources, flag_value=8):
+    ny, nx = image.shape
+    # Check 4 points on the kron aperture, one on each side of the major and minor axis
+    minor_xmin = sources['x'] - sources['b'] * sources['kronrad'] * np.sin(np.deg2rad(sources['theta']))
+    minor_xmax = sources['x'] + sources['b'] * sources['kronrad'] * np.sin(np.deg2rad(sources['theta']))
+    minor_ymin = sources['y'] - sources['b'] * sources['kronrad'] * np.cos(np.deg2rad(sources['theta']))
+    minor_ymax = sources['y'] + sources['b'] * sources['kronrad'] * np.cos(np.deg2rad(sources['theta']))
+    major_ymin = sources['y'] - sources['a'] * sources['kronrad'] * np.sin(np.deg2rad(sources['theta']))
+    major_ymax = sources['y'] + sources['a'] * sources['kronrad'] * np.sin(np.deg2rad(sources['theta']))
+    major_xmin = sources['x'] - sources['a'] * sources['kronrad'] * np.cos(np.deg2rad(sources['theta']))
+    major_xmax = sources['x'] + sources['a'] * sources['kronrad'] * np.cos(np.deg2rad(sources['theta']))
+
+    # Note we are already 1 indexed here
+    sources_off = np.logical_or(minor_xmin < 1, major_xmin < 1)
+    sources_off = np.logical_or(sources_off, minor_ymin < 1)
+    sources_off = np.logical_or(sources_off, major_ymin < 1)
+    sources_off = np.logical_or(sources_off, minor_xmax > nx)
+    sources_off = np.logical_or(sources_off, major_xmax > nx)
+    sources_off = np.logical_or(sources_off, minor_ymax > ny)
+    sources_off = np.logical_or(sources_off, major_ymax > ny)
+    sources[sources_off] |= flag_value
 
 
 class SourceDetector(Stage):
-    # Note that threshold is number of sigma, not an absolute number because we provide the error
-    # array to SEP.
-    threshold = 10.0
+    threshold = 2.5
```

Comment:
where did this threshold come from? Is it no longer a number of sigma but an absolute number?